### PR TITLE
NAS-132604 / 25.04 / Add support for ipv6 in docker plugin

### PR DESCRIPTION
**Pull request by Qubad786**

### CODE REVIEW

File: src/middlewared/middlewared/api/v25_04_0/docker.py [hunk 1]
```diff
@@ -9,14 +10,25 @@
...
     @field_validator('base')
+    @classmethod
     def check_prefixlen(cls, v):
         if v.network.prefixlen in (32, 128):
             raise ValueError('Prefix length of base network cannot be 32 or 128.')
         return v
 
+    @model_validator(mode='after')
+    def validate_attrs(self):
+        if isinstance(self.base, IPv4Interface):
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/docker.py:24]: Why do we need to do `isinstance` check here? Can we not do
```
if self.base.version == 4 and self.size > 32:
```

File: src/middlewared/middlewared/api/v25_04_0/docker.py [hunk 2]
```diff
@@ -25,6 +35,16 @@ class DockerEntry(BaseModel):
     pool: NonEmptyString | None
     nvidia: bool
     address_pools: list[AddressPool]
+    cidr_v6: IPvAnyInterface
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/docker.py:38]: Why would this be of type `IPvAnyInterface` if we just accept an ipv6 address?

Comment (Qubad786) [reply src/middlewared/middlewared/api/v25_04_0/docker.py:38]: @yocalebo validation for this has been added below - we don't have a type from pydantic which highlights ipv4/ipv6 interfaces separately or do you think we should perhaps use `ipaddress` directly in this case?

Comment (yocalebo) [reply src/middlewared/middlewared/api/v25_04_0/docker.py:38]: We don't want to use pydantic field types if we don't have to. Especially if there is a type that is builtin to python. So yeah, specifing `ipaddress.IPv6*` type is preferred here.

**Review by yocalebo — APPROVED** [review]

**Pull request merged**